Remaining issues (#152)
NTAP/rfc8312bis: draft-ietf-tcpm-rfc8312bis.md

    - ins: Sangtae Ha
    target: "https://web.archive.org/web/20161118125842/http://netsrv.csc.ncsu.edu/wiki/index.php/TCP_Testing"

  AIMD-friendliness:
    title: Friendliness between AIMD Algorithms
    date: 2022-08-08
    author:

>     - ins: Bob Briscoe
>     - ins: Olga Albisser

Reviewer: ```suggestion
    - ins: B. Briscoe
    - ins: O. Albisser
```

obsoleting {{?RFC8312}}. This requires an update to {{!RFC5681}}, which
limits the aggressiveness of Reno TCP implementations in its Section 3.
Since CUBIC is occasionally more aggressive than the {{!RFC5681}}
algorithms, this document updates the first paragraph of Section 3 of
{{!RFC5681}}, replacing it with a normative reference to guideline (1)

> in Section 3 of {{!RFC5033}}, which allows for CUBIC's behavior defined

Reviewer: ```suggestion
in Section 3 of {{!RFC5033}}, which allows for CUBIC's behavior as defined
```

> obsoleting {{?RFC8312}}. This requires an update to {{!RFC5681}}, which
> limits the aggressiveness of Reno TCP implementations in its Section 3.

Reviewer: ```suggestion
obsoleting {{?RFC8312}}. This requires an update to {{Section3 of !RFC5681}}, which
limits the aggressiveness of Reno TCP implementations.
```

Since CUBIC is occasionally more aggressive than the {{!RFC5681}}

> algorithms, this document updates the first paragraph of Section 3 of
> {{!RFC5681}}, replacing it with a normative reference to guideline (1)

Reviewer: ```suggestion
algorithms, this document updates the first paragraph of {{Section 3 of
!RFC5681}}, replacing it with a normative reference to guideline (1)
```

> in Section 3 of {{!RFC5033}}, which allows for CUBIC's behavior as defined

Reviewer: ```suggestion
in {{Section 3 of !RFC5033}}, which allows for CUBIC's behavior as defined
```

~~~
{: #eq3 artwork-align="center" }

> By the same analysis, to achieve similar average window size as Reno

Reviewer: ```suggestion
By the same analysis, to achieve a similar average window size as Reno
```

~~~
{: artwork-align="center" }

which achieves approximately the same average window size as Reno in

> many cases. The model used to calculate the alpha_cubic value is not

Reviewer: ```suggestion
many cases. The model used to calculate {{{α}{}}}*<sub>cubic</sub>* is not
```

absolutely precise, but analysis and simulation {{AIMD-friendliness}},
as well as over a decade of experience with CUBIC in the public
Internet, shows that this approach produces acceptable levels of

> rate fairness between CUBIC and Reno flows. When

Reviewer 2: I might want to add some nuance for future explorations on this point. How about something like this?
 
'.. shows that this approach produces acceptable levels of rate fairness between CUBIC and Reno flows. Also, no significant drawbacks for the model has been reported up to this point. However, It would be beneficial to have further detailed analysis on it'

Author: Applied Lars' text that was based on the above suggestion.

> rate fairness between CUBIC and Reno flows. When

Reviewer: ```suggestion
rate fairness between CUBIC and Reno flows. Also, no significant
drawbacks of the model have been reported. However, It would be
beneficial to see continued detailed analysis on it. When
```
Suggested rephrasing of @nsdyoshi's comment.

~~~
{: artwork-align="center" }

which achieves approximately the same average window size as Reno in
many cases. The model used to calculate {{{α}{}}}*<sub>cubic</sub>* is not
absolutely precise, but analysis and simulation {{AIMD-friendliness}},
as well as over a decade of experience with CUBIC in the public

> Internet, shows that this approach produces acceptable levels of

Reviewer: ```suggestion
Internet, show that this approach produces acceptable levels of
```

rate fairness between CUBIC and Reno flows. Also, no significant

> drawbacks of the model have been reported. However, It would be

Reviewer: ```suggestion
drawbacks of the model have been reported. However, it would be
```

{: #eqssthresh artwork-align="center" }

A side effect of setting {{{β}{}}}*<sub>cubic</sub>* to a value bigger

> than 0.5 is that packet loss can happen for more than 1 round-trips in certain

Reviewer: ```suggestion
than 0.5 is that packet loss can happen for more than one round-trip in certain
```

cases, but it can work efficiently in other cases, for example, when HyStart++

> is used along with CUBIC or when sending rate is limited by the application.

Reviewer: ```suggestion
is used along with CUBIC or when the sending rate is limited by the application.
```

We believe that while a more adaptive setting of {{{β}{}}}*<sub>cubic</sub>*

> could help to limit packet loss to a single round, it will make the analysis

Reviewer 3: Perhaps "it would make the analysis of CUBIC much harder" rather than "it will make the analysis of CUBIC much harder"?

Reviewer 2: How about  "it would require detailed analysis and large-scale evaluations to validate such algorithms"

Author: Applied Lars' suggestion which combines both feedbacks.

> could help to limit packet loss to a single round, it will make the analysis
> of CUBIC much harder.

Reviewer: ```suggestion
could help to limit packet loss to a single round, it would require detailed
analyses and large-scale evaluations to validate such algorithms.
```
@nsdyoshi suggestion.

{: #eqssthresh artwork-align="center" }

A side effect of setting {{{β}{}}}*<sub>cubic</sub>* to a value bigger
than 0.5 is that packet loss can happen for more than one round-trip in certain
cases, but it can work efficiently in other cases, for example, when HyStart++
is used along with CUBIC or when the sending rate is limited by the application.

> We believe that while a more adaptive setting of {{{β}{}}}*<sub>cubic</sub>*

Reviewer: ```suggestion
While a more adaptive setting of {{{β}{}}}*<sub>cubic</sub>*
```

> could help to limit packet loss to a single round, it would require detailed

Reviewer: ```suggestion
could help limit packet loss to a single round, it would require detailed
```

and CUBIC to achieve a certain throughput. We use 1500-byte packets
and an *RTT* of 0.1 seconds.

> Our test results in {{HLRX07}} indicate that, in typical cases with a

Reviewer: ```suggestion
The test results in {{HLRX07}} indicate that, in typical cases with a
```
